### PR TITLE
angular npm run test should pass

### PR DESCRIPTION
**Pull request by Tanya0609**
This PR is to address the tests suggestions from #681 

Task:https://microsoftgarage.visualstudio.com/Intern%20GitHub/_sprints/taskboard/WTS%20Team/Intern%20GitHub/Y19-VS/05
Task:https://microsoftgarage.visualstudio.com/Intern%20GitHub/_sprints/taskboard/WTS%20Team/Intern%20GitHub/Y19-VS/05

(Since the task "master detail page has error" causes three of the tests failing, so I put the fix for that error in this PR too)

**My fix for test files:**
In Angular component test files, if we used some other components or modules in that component, than we need to "declare" those components and "import" the modules in the test file too. 

**My fix for the error:**
This error happened because one variable in master detail page did not get initialized, thus I add the initialization.

**How to test:**
1. generare a angular project with 4 types of pages selected
2. do npm install
3. do npm run test, all tests should all pass ( I selected 4 pages, thus 17 tests)
ng-test

### CODE REVIEW

**Review by jsondoo — APPROVED** [review]
All tests passing 👍

**Review by streetcoder123 — APPROVED** [review]

File: templates/Web/Pages/Angular.MasterDetail/src/app/app-shell/Param_SourceName_Kebab/master-detail.service.ts [hunk 1]
```diff
@@ -19,13 +19,13 @@ export class MasterDetailService {
 }
 
 export interface IMasterDetailText {
-  paragraph: string;
   title: string;
   tabName: string;
   id: number;
   status: string;
-  orderDate: Date;
+  orderDate: string;
```
Comment (SahilTara) [inline templates/Web/Pages/Angular.MasterDetail/src/app/app-shell/Param_SourceName_Kebab/master-detail.service.ts:26]: why the change of orderDate to string? Just inquiring.

File: templates/Web/Pages/Angular.Grid/src/app/app-shell/Param_SourceName_Kebab/grid-box/grid-box.component.spec.ts [hunk 2]
```diff
@@ -1,4 +1,5 @@
 ﻿import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { MatIconModule } from '@angular/material/icon';
```
Comment (SahilTara) [inline templates/Web/Pages/Angular.Grid/src/app/app-shell/Param_SourceName_Kebab/grid-box/grid-box.component.spec.ts:2]: Hmm do we need material's icon module? 
Refer to: #766
There are other uses of this in this PR as well.

File: templates/Web/Pages/Angular.MasterDetail/src/app/app-shell/Param_SourceName_Kebab/master-detail.service.ts (hunk 1 above)
Comment (Tanya0609) [reply templates/Web/Pages/Angular.MasterDetail/src/app/app-shell/Param_SourceName_Kebab/master-detail.service.ts:26]: cause when they use orderDate, it was orderDdate  = new Date().**toDateString**; thus it is a string actually

File: templates/Web/Pages/Angular.Grid/src/app/app-shell/Param_SourceName_Kebab/grid-box/grid-box.component.spec.ts (hunk 2 above)
Comment (Tanya0609) [reply templates/Web/Pages/Angular.Grid/src/app/app-shell/Param_SourceName_Kebab/grid-box/grid-box.component.spec.ts:2]: We are going to remove all material design from generated code. Thus, we are going to remove all material design code ( including this one) from generated code and other styling change in another PR #766.

**Review by SahilTara — APPROVED** [review]

**Review by piotrmark — APPROVED** [review]
